en/22/10.md

```diff
@@ -0,0 +1,291 @@
+          multiversx_sc::imports!();
+          multiversx_sc::derive_imports!();
+
+          #[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, TypeAbi)]
+          pub struct Kitty {
+              pub is_gestating: bool,
+              pub is_ready: bool,
+              pub cooldown_index: u64,
+              pub next_action_at: u64,
+              pub siring_with_id: u64,
+              pub birth_time: u64,
+              pub matron_id: u64,
+              pub sire_id: u64,
+              pub generation: u64,
+              pub genes: u64,
+          }
+
+          #[multiversx_sc::proxy]
+          pub trait CryptoKitties {
+              #[endpoint]
+              fn get_kitty(&self, id: usize) -> Kitty;
+          }
+      }
+
+      #[multiversx_sc::module]
+      pub trait ZombieFeeding: storage::Storage + zombie_factory::ZombieFactory {
+          #[endpoint]
+          fn feed_and_multiply(&self, zombie_id: usize, target_dna: u64) {
+              let caller = self.blockchain().get_caller();
+              require!(
+                  caller == self.zombie_owner(&zombie_id).get(),
+                  "Only the owner of the zombie can perform this operation"
+              );
+              let my_zombie = self.zombies(&zombie_id).get();
+              let dna_digits = self.dna_digits().get();
+              let max_dna_value = u64::pow(10u64, dna_digits as u32);
+              let verified_target_dna = target_dna % max_dna_value;
+              let new_dna = (my_zombie.dna + verified_target_dna) / 2;
+              self.create_zombie(caller, ManagedBuffer::from(b"NoName"), new_dna);
+          }
+
+          #[proxy]
+          fn kitty_proxy(&self, to: ManagedAddress) -> crypto_kitties_proxy::Proxy<Self::Api>;
+      }
+---
+
+It's time to feed our zombies! And what do zombies like to eat most?
+
+Well it just so happens that CryptoZombies love to eat...
+
+**CryptoKitties!** 😱😱😱
+
+(Yes, I'm serious 😆 )
+
+In order to do this we'll need to read the kittyDna from the CryptoKitties smart contract. We can do that because the CryptoKitties data is stored openly on the blockchain. Isn't the blockchain cool?!
+
+Don't worry — our game isn't actually going to hurt anyone's CryptoKitty. We're only *reading* the CryptoKitties data, we're not able to actually delete it 😉
+
+## Interacting with other contracts
+
+For our contract to talk to another contract on the blockchain that we don't own, first we need to define a **_Proxy_**.
+
+Let's look at a simple example. Say there was a contract on the blockchain that looked like this:
+
+```
+#[multiversx_sc::contract]
+pub trait Adder {
+   
+    ...
+
+    #[endpoint]
+    fn add(&self, a: BigUint, b: BigUint) -> BigUint {
+        a + b
+    }
+}
+```
+
+This would be a simple contract where you would sum up 2 numbers and returns their sum.
+
+Now let's say we had an external contract that wanted to read the data in this contract using the `add` function. 
```

Comment:
Not sure that reading data in this contract is technically correct. Please double-check.